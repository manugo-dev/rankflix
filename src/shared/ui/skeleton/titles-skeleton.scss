.titles-skeleton {
  overflow: hidden;
}

.titles-skeleton__track {
  display: grid;
  overflow: hidden;
  padding: 1rem;
  gap: 1rem;
  grid-auto-columns: minmax(12.5rem, 20%);
  grid-auto-flow: column;
}

.titles-skeleton__card {
  flex-shrink: 0;
  border-radius: 0.75rem;
  animation: shimmer 1.6s infinite linear;
  aspect-ratio: 2 / 3;
  background: linear-gradient(90deg, #2a2a2a 25%, #3a3a3a 50%, #2a2a2a 75%);
  background-size: 200% 100%;
}

@keyframes shimmer {
  from {
    background-position: 200% 0;
  }

  to {
    background-position: -200% 0;
  }
}
