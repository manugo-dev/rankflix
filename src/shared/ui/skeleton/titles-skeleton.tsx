import "./titles-skeleton.scss";

export function TitlesSkeleton() {
  return (
    <div className="titles-skeleton">
      <div className="titles-skeleton__track">
        {Array.from({ length: 10 }).map((_, index) => (
          <div key={index} className="titles-skeleton__card" />
        ))}
      </div>
    </div>
  );
}
