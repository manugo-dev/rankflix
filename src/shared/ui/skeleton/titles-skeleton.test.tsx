import { cleanup, render } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

import { TitlesSkeleton } from "./titles-skeleton";

const DEFAULT_PROPS = {};

function renderTitlesSkeleton() {
  return render(<TitlesSkeleton {...DEFAULT_PROPS} />);
}

describe("TitlesSkeleton", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  afterEach(() => {
    cleanup();
  });

  it("matches the snapshot", () => {
    const { container } = renderTitlesSkeleton();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders ten skeleton cards", () => {
    renderTitlesSkeleton();

    const cards = document.querySelectorAll(".titles-skeleton__card");

    expect(cards.length).toBe(10);
  });

  it("renders the outer track container", () => {
    renderTitlesSkeleton();

    const track = document.querySelector(".titles-skeleton__track");

    expect(track).toBeTruthy();
  });
});
