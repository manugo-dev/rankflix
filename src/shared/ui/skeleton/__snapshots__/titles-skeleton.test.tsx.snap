// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitlesSkeleton > matches the snapshot 1`] = `
<div
  class="titles-skeleton"
>
  <div
    class="titles-skeleton__track"
  >
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
    <div
      class="titles-skeleton__card"
    />
  </div>
</div>
`;
